src/lib/components/Checkbox/root.svelte

```diff
@@ -0,0 +1,100 @@
+<script lang="ts" context="module">
+	import { controllableState } from '$lib/helpers/controllableState';
+	import { uniqueContext } from '$lib/helpers/uniqueContext';
+	import type { BaseProps } from '$lib/types';
+	import { derived, type Readable, type Writable } from 'svelte/store';
+
+	export type Props = BaseProps<HTMLButtonElement> & {
```

Comment:
I'm curious what situations you are thinking of where they are useful (and where `ComponentProps` wouldn't work)

Reply:
If someone wants to extend the Checkbox Component for any reason, and wants to use restProp. ComponentProps is only a react thing, right?

Reply:
No, ComponentProps is Svelte, pulls out the props of a svelte component. Thats how I made the docs helper type safe.

Reply:
Oh, TIL. This is fine then!

Reply:
Try `ComponentProps<InstanceType<typeof Collapsible.Content>>>` (a bit lengthy, but i believe how its done in the helper).

Reply:
That works! I think it can be a bit obtuse, maybe we could export a RadixComponentProps that basically just wraps that around, something like that. But doesn't need to be in this PR.